bridges/otelzap/encoder.go

```diff
@@ -196,15 +194,13 @@ type arrayEncoder struct {
 }
 
 func (a *arrayEncoder) AppendArray(v zapcore.ArrayMarshaler) error {
-	// TODO: Use arrayEncoder from a pool.
 	arr := &arrayEncoder{}
 	err := v.MarshalLogArray(arr)
 	a.elems = append(a.elems, log.SliceValue(arr.elems...))
 	return err
 }
 
 func (a *arrayEncoder) AppendObject(v zapcore.ObjectMarshaler) error {
-	// TODO: Use objectEncoder from a pool.
 	m := newObjectEncoder(2)
```

Comment:
It is always `2`. Maybe we can move the magic value inside `newObjectEncoder` (and just have `newobj := newObjectEncoder()` and also explain why it is 2 and not any different value.

This can be handled in a separate PR.